src/Http/RequestFactory.php

```diff
@@ -194,30 +194,54 @@ public function createHttpRequest()
 		$usingTrustedProxy = $remoteAddr && array_filter($this->proxies, function ($proxy) use ($remoteAddr) {
 			return Helpers::ipMatch($remoteAddr, $proxy);
 		});
-
 		if ($usingTrustedProxy) {
-			if (!empty($_SERVER['HTTP_X_FORWARDED_PROTO'])) {
-				$url->setScheme(strcasecmp($_SERVER['HTTP_X_FORWARDED_PROTO'], 'https') === 0 ? 'https' : 'http');
-			}
+			if(!empty($_SERVER['HTTP_FORWARDED'])) {
+				$forwardParams = preg_split('/[,]|[;]/', $_SERVER['HTTP_FORWARDED']);
```

Comment:
It is not. However it should not be a problem, as I believe there is no need for the distinction.
There two headers would produce the same result array:
for=192.0.2.43, for=198.51.100.17
for=192.0.2.43; for=198.51.100.17

I just need to split the "key-value" pairs and put them in the proper sub-array.